src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/AppxModuleHelper.cs

```diff
@@ -319,53 +330,69 @@ private async Task AddAppInstallerBundleAsync(string releaseTag, bool downgrade,
-                GetAppxPackage,
-                new Dictionary<string, object>
-                {
-                    { Name, VCLibsUWPDesktop },
-                });
+            Dictionary<string, string> appxPackages = new Dictionary<string, string>();
+            var arch = RuntimeInformation.OSArchitecture;
 
-            // See if the minimum (or greater) version is installed.
-            // TODO: Pull the minimum version from the target package
-            Version minimumVersion = new Version(VCLibsUWPDesktopVersion);
+            string appxPackageX64 = string.Format(AppxPackageTemplate, dependencies.Name, dependencies.Version, "x64");
+            string appxPackageX86 = string.Format(AppxPackageTemplate, dependencies.Name, dependencies.Version, "x86");
+            string appxPackageArm = string.Format(AppxPackageTemplate, dependencies.Name, dependencies.Version, "arm");
+            string appxPackageArm64 = string.Format(AppxPackageTemplate, dependencies.Name, dependencies.Version, "arm64");
 
-            // Construct the list of frameworks that we want present.
-            Dictionary<string, string> vcLibsDependencies = new Dictionary<string, string>();
-            var arch = RuntimeInformation.OSArchitecture;
             if (arch == Architecture.X64)
             {
-                vcLibsDependencies.Add("x64", VCLibsUWPDesktopX64);
+                appxPackages.Add("x64", appxPackageX64);
             }
             else if (arch == Architecture.X86)
             {
-                vcLibsDependencies.Add("x86", VCLibsUWPDesktopX86);
+                appxPackages.Add("x86", appxPackageX86);
             }
             else if (arch == Architecture.Arm64)
             {
                 // Deployment please figure out for me.
-                vcLibsDependencies.Add("x64", VCLibsUWPDesktopX64);
-                vcLibsDependencies.Add("x86", VCLibsUWPDesktopX86);
-                vcLibsDependencies.Add("arm", VCLibsUWPDesktopArm);
-                vcLibsDependencies.Add("arm64", VCLibsUWPDesktopArm64);
+                appxPackages.Add("x64", appxPackageX64);
+                appxPackages.Add("x86", appxPackageX86);
+                appxPackages.Add("arm", appxPackageArm);
+                appxPackages.Add("arm64", appxPackageArm64);
             }
             else
             {
                 throw new PSNotSupportedException(arch.ToString());
             }
 
+            return appxPackages;
+        }
+
+        private void FindMissingDependencies(Dictionary<string, string> dependencies, string packageName, string requiredVersion)
+        {
+            var result = this.ExecuteAppxCmdlet(
+                GetAppxPackage,
+                new Dictionary<string, object>
+                {
+                    { Name, packageName },
+                });
+
+            // See if the minimum (or greater) version is installed.
+            // TODO: Pull the minimum version from the target package

```

Comment:
I would remove this TODO; we are intentionally not doing that now because it is very expensive to read the manifest from a remote package as compared to the small json file we are creating.